Validator/CorsValidator.php

```diff
@@ -62,14 +62,21 @@ private function configurationIsWildcard(): bool
      */
     public function originIsValid(): bool
     {
+        $isValid = false;
         if ($this->request instanceof HttpRequest) {
             //Validate that we're working with something that cares about CORS
-            if (!$this->request->getHeader('Origin')) {
-                return false;
-            }
+            try {
+                if (!$this->request->getHeader('Origin')) {
```

Comment:
I think it would make sense to extract `!$this->request->getHeader('Origin')` from:

```
if(!$this->request->getHeader('Origin')){

}
```

To a new private function, perhaps called: `hasOriginHeader`

```php
if(!$this->hasOriginHeader())
{
 return false;
}
```

And you would swallow the Zend error in that function and return false.

It keeps the concerns of this specific use-case outside the core validation function.

Reply:
:+1: done